Explorer/Assets/DCL/Notifications/NotificationsRequestController.cs

```diff
@@ -118,24 +119,48 @@ await webRequestController.GetAsync(
                 if (notifications.Count > 0)
                     lastPolledTimestamp = DateTime.UtcNow.UnixTimeAsMilliseconds();
 
-                await UniTask.WhenAll(notifications.Select(notification =>
-                {
-                    notificationsBusController.AddNotification(notification);
-                    return SetNotificationAsReadAsync(notification.Id, ct);
-                }));
+
+                using var scope = ThreadSafeListPool<string>.SHARED.Get(out var list);
+                foreach (INotification notification in notifications)
+                    try
+                    {
+                        notificationsBusController.AddNotification(notification);
+                        list.Add(notification.Id);
+                    }
+                    catch (Exception e)
+                    {
+                        ReportHub.LogException(e, ReportCategory.UI);
+                    }
+
+                await SetNotificationAsReadAsync(list, ct);
             }
             while (ct.IsCancellationRequested == false);
         }
 
         public async UniTask SetNotificationAsReadAsync(string notificationId, CancellationToken ct)
+        {
+            using var scope = ThreadSafeListPool<string>.SHARED.Get(out var list);
+            list.Add(notificationId);
+            await SetNotificationAsReadAsync(list, ct);
```

Comment:
Personally, I would just do separate implementations. One for single notification, and one for batch.